server/src/main/java/org/opensearch/rest/action/search/RestCreatePitAction.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.search;
+
+import org.opensearch.action.search.CreatePitRequest;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.Strings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * Rest action for creating PIT context
+ */
+public class RestCreatePitAction extends BaseRestHandler {
+    public static String ALLOW_PARTIAL_PIT_CREATION = "allow_partial_pit_creation";
+    public static String KEEP_ALIVE = "keep_alive";
+
+    @Override
+    public String getName() {
+        return "create_pit_action";
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        boolean allowPartialPitCreation = request.paramAsBoolean(ALLOW_PARTIAL_PIT_CREATION, true);
+        String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
+        TimeValue keepAlive = request.paramAsTime(KEEP_ALIVE, null);
+        CreatePitRequest createPitRequest = new CreatePitRequest(keepAlive, allowPartialPitCreation, indices);
+        createPitRequest.setIndicesOptions(IndicesOptions.fromRequest(request, createPitRequest.indicesOptions()));
+        createPitRequest.setPreference(request.param("preference"));
+        createPitRequest.setRouting(request.param("routing"));
+
+        return channel -> client.createPit(createPitRequest, new RestStatusToXContentListener<>(channel));
```

Comment:
What happens is the client connection gets closed before PIT can be returned back? Although the client can do a list maybe to verify but most clients use the default retry strategy that wouldn't do a list, will we end up retaining PITs espl if keep alive was sufficiently long

Reply:
Yes, PIT will get created in the backend. I looked at other rest actions, don't see any specific handling for such scenarios - are we looking for any specific implementation ?

Reply:
Should we cleanup created PIT if client connection closes before it receives the PIT id
cc: @reta for thoughts

Reply:
That's an interesting one, we need an `pit_id` to cleanup, but as you said, if client closes the connection before PIT id received, not much we can do. However, we could follow other APIs (fe `RestSearchScrollAction`) and support `POST  "/{index}/_search/point_in_time/${pit_id}")` - in this case the `pit_id` is coming from client and could be managed in idempotent way, what do you think @Bukhtawar ?

Reply:
@reta Not sure if I understood it correctly. In the current context, first a PIT needs to be created using this API and then used in subsequent search APIs. If the client closes the connection, just like search cancellation cleans up ongoing searches, I was thinking if we could clean up(delete) the PIT or release the underlying Searcher. Thoughts?

Reply:
@Bukhtawar oh got it, you mean on the server, right? Deleting PIT would probably be too agressive (if client reconnects and try to use it again), but releasing underlying Searcher would make sense (since client won't be able to fetch the results anyway).